internal/reflect_utils/reflect_utils.go

```diff
@@ -0,0 +1,25 @@
+package reflect_utils
```

Comment:
I hate the name `utils` in packages... see https://go.dev/wiki/CodeReviewComments#package-names. If we don't want to clash with `reflect` from Go's standard library, let's call this `tfreflect` instead.